src/runtime/SharedArrayBufferObject.cpp

```diff
@@ -29,6 +29,21 @@
 
 namespace Escargot {
 
+static void backingStoreDeleter(void* data, size_t length, void* deleterData)
+{
+    SharedArrayBufferObjectBackingStoreData* internalData = (SharedArrayBufferObjectBackingStoreData*)deleterData;
+    internalData->deref();
+}
+
+void SharedArrayBufferObjectBackingStoreData::deref()
+{
+    m_refCount--;
+    if (m_refCount == 0) {
+        Global::platform()->onFreeArrayBufferObjectDataBuffer(m_data, m_byteLength);
+        delete this;
```

Comment:
`m_refCount` is guaranteed for its atomic operation.
IMHO, but it seems possible that this delete operation could be run simultaneously when several threads go into if statement at the same time (I'm not sure its memory order)
What do you think about it?

Reply:
Good point!
I didn't imagine that.
I am going to fix problem right now

Reply:
Looks great 🚀